src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -397,6 +397,24 @@ void Terminal::Write(std::wstring_view stringView)
     _stateMachine->ProcessString(stringView);
 }
 
+void Terminal::WritePastedText(std::wstring_view stringView)
+{
+    auto option = ::Microsoft::Console::Utils::FilterOption::CarriageReturnNewline |
+                  ::Microsoft::Console::Utils::FilterOption::ControlCodes;
+
+    std::wstring filtered = ::Microsoft::Console::Utils::FilterStringForPaste(stringView, option);
+    if (IsXtermBracketedPasteModeEnabled())

```

Comment:
here's an idea. Maybe to handle Jantari's case (since they want to _paste literal control sequences_), we can strip control codes only when bracketed paste mode is enabled.

@j4james, do you think that's a reasonable compromise? It solves the case where a clipboard payload can break out of bracketed paste and doesn't make unguarded paste any worse.

At the same time, I still can't see the value in pasting raw control codes here today in Common Era 2021, so...

The only time I ever did it was when I was actively trying to blow up bracketed paste in xterm/rxvt.

Otherwise, that's what `^V` is for.

Reply:
Personally, my vote is let's strip them always.

Reply:
I'd like to just strip them always, too. Also I don't like the idea of coupling "control codes filtering" and "bracketed paste mode". This is what @j4james told me specifically not to do in https://github.com/microsoft/terminal/pull/7508#discussion_r482996989

Reply:
For the record, I'm in agreement with you guys. I think it's preferable that they're always filtered.